x-pack/plugins/data_visualizer/public/application/data_comparison/constants.ts

```diff
@@ -0,0 +1,1047 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+
+export const DATA_COMPARISON_TYPE = {
+  NUMERIC: 'numeric',
+  CATEGORICAL: 'categorical',
+  UNSUPPORTED: 'unsupported',
+} as const;
+
+export const NUMERIC_TYPE_LABEL = i18n.translate(
+  'xpack.dataVisualizer.dataComparison.numericLabel',
+  {
+    defaultMessage: 'Numeric',
+  }
+);
+export const CATEGORICAL_TYPE_LABEL = i18n.translate(
+  'xpack.dataVisualizer.dataComparison.categoricalLabel',
+  {
+    defaultMessage: 'Categorical',
+  }
+);
+
+export const UNSUPPORTED_LABEL = i18n.translate(
+  'xpack.dataVisualizer.dataComparison.UnsupportedLabel',
+  {
+    defaultMessage: 'Unsupported',
+  }
+);
+
+export const REFERENCE_LABEL = i18n.translate(
```

Comment:
For a generic data comparison view, I'm leaning towards using the same terms as we have in the Explain Log Rate Spikes page, `Baseline` and `Deviation`, or maybe `Reference` and `Deviation`.